plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/PluginModernizer.java

```diff
@@ -1,23 +1,142 @@
+        File pluginsDir = new File(projectRoot, "test-plugins");
+
+        File[] subdirectories = pluginsDir.listFiles(File::isDirectory);
+        if (subdirectories != null) {
+            for (File pluginDir : subdirectories) {
+                try {
+                    List<String> goals = createGoalsList();
+                    if (goals == null) {
+                        LOG.info("No active recipes.");
+                        return;
+                    }
+                    invokeMavenGoals(pluginDir, goals);
+                } catch (IOException e) {
+                    LOG.error(e.getMessage(), e);
+                }
+            }
+        } else {
+            LOG.error("No subdirectories found in {}", pluginsDir.getAbsolutePath());
+        }
         LOG.info("Plugins: {}", config.getPlugins());
         LOG.info("Recipes: {}", config.getRecipes());
         LOG.debug("Cache Path: {}", config.getCachePath());
     }
 
+    private static void invokeMavenGoals(File pluginDir, List<String> goals) {
+        Invoker invoker = new DefaultInvoker();
+        String mavenHome = getMavenHomePath();
+        if (mavenHome == null) {
+            LOG.error("Neither MAVEN_HOME nor M2_HOME environment variables are set.");
+            return;
+        }
+        try {
+            Path mavenHomePath = Paths.get(mavenHome).toRealPath();
+            if (!Files.isDirectory(mavenHomePath) || !Files.isExecutable(mavenHomePath.resolve("bin/mvn"))) {
+                LOG.error("Invalid Maven home directory. Aborting build.");
+                return;
+            }
+            invoker.setMavenHome(new File(mavenHome));
+            InvocationRequest request = new DefaultInvocationRequest();
+            request.setPomFile(new File(pluginDir, "pom.xml"));
+            request.setGoals(goals);
+            InvocationResult result = invoker.execute(request);
+            if (result.getExitCode() != 0) {
+                LOG.error("Build failed.");
+                if (result.getExecutionException() != null) {
+                    LOG.error("Execution exception occurred: ", result.getExecutionException());
+                }
+            } else {
+                LOG.info("Build succeeded.");
+            }
+        } catch (MavenInvocationException e) {
+            LOG.error("Maven invocation failed: ", e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static String getMavenHomePath() {
+        String mavenHome = System.getenv("MAVEN_HOME");
+        if (mavenHome == null) {
+            mavenHome = System.getenv("M2_HOME");
+        }
+        return mavenHome;
+    }
+
+    List<String> createGoalsList() throws IOException {
+        List<String> goals = new ArrayList<>();
+        goals.add("org.openrewrite.maven:rewrite-maven-plugin:run");
+
+        try (InputStream inputStream = PluginModernizer.class.getResourceAsStream("/recipe_data.json")) {
```

Comment:
I would extract this logic a a separate class that PluginModernizer.java is using. 

It the the PluginModernizer.java will just call using and orchestrate the modernizing of plugin